luigi_pipeline/tests/test_seqr_gcnv_loading.py

```diff
@@ -0,0 +1,394 @@
+            data = data.drop(k)
+    return data
+
+class SeqrGCNVGeneParsingTest(unittest.TestCase):
+
+    def test_parse_genes(self):
+        t1 = hl.Table.parallelize(
+            [
+                {"genes": "AC118553.2,SLC35A3"}, 
+                {"genes": "AC118553.1,None"},
+                {"genes": "None"},
+                {"genes": "SLC35A3.43"},
+                {"genes": ""},
+                {"genes": "SLC35A4.43"},
+            ], 
+            hl.tstruct(genes=hl.dtype('str')), 
+            key="genes"
+        )
+        t1 = t1.annotate(gene_set = parse_genes(t1.genes))
+        self.assertCountEqual(
+            t1.collect(),
+            [
+                hl.Struct(genes="AC118553.2,SLC35A3", gene_set=["AC118553", "SLC35A3"]),
+                hl.Struct(genes="AC118553.1,None", gene_set=["AC118553"]),
+                hl.Struct(genes="None", gene_set=[]),
+                hl.Struct(genes="SLC35A3.43", gene_set=["SLC35A3"]),
+                hl.Struct(genes="", gene_set=[]),
+                hl.Struct(genes="SLC35A4.43", gene_set=["SLC35A4"]),
+            ],
+        )
+
+    def test_aggregate_parsed_genes(self):
+        t1 = hl.Table.parallelize(
+            [
+                {"genes": "AC118553.2,SLC35A3"}, 
+                {"genes": "AC118553.1,None"}, 
+                {"genes": "None"},
+                {"genes": "SLC35A3.43"}, 
+                {"genes": ""}, 
+                {"genes": "SLC35A4.43"},
+            ], 
+            hl.tstruct(genes=hl.dtype('str')), 
+            key="genes"
+        )
+        aggregated_gene_set = t1.aggregate(
+            hl_agg_collect_set_union(parse_genes(t1.genes))
+        )
+        self.assertEqual(
+            aggregated_gene_set,
+            set(["SLC35A4", "SLC35A3", "AC118553"])
+        )
+
+class SeqrGCNVLoadingTest(unittest.TestCase):
+
+    def setUp(self):
+        self._temp_dir = tempfile.TemporaryDirectory()
+        self._variant_mt_file = tempfile.mkstemp(dir=self._temp_dir.name, suffix='.mt')[1]
+        self._genotypes_mt_file = tempfile.mkstemp(dir=self._temp_dir.name, suffix='.mt')[1]
+
+    def tearDown(self):
+        shutil.rmtree(self._temp_dir.name)
+
+    @mock.patch('lib.model.gcnv_mt_schema.datetime', wraps=datetime)
+    @mock.patch('lib.hail_tasks.HailElasticsearchClient')
+    def test_run_new_joint_tsv_task(self, mock_elasticsearch_client, mock_datetime):
+        mock_datetime.date.today.return_value = datetime.date(2022, 12, 2)
+        worker = luigi.worker.Worker()
+        variant_task = SeqrGCNVVariantMTTask(
+            source_paths=NEW_JOINT_CALLED_CALLSET,
+            dest_path=self._variant_mt_file,
+        )
+        genotype_task = SeqrGCNVGenotypesMTTask(
+            genome_version="38",
+            source_paths="i am completely ignored",
+            dest_path=self._genotypes_mt_file,
+            is_new_joint_call=True,
+        )
+        export_task = SeqrGCNVMTToESTask()
+        export_task._es.reset_mock()
```

Comment:
I know export_task._es has been used when calling `SeqrGCNVMTToESTask()` above. But is it necessary to do `reset_mock()` here? To me, you can remove this line.

Reply:
I fixed this!  The source of the bug was luigi re-using instances of `SeqrGCNVMTToESTask`.  I didn't know that task instances with the same parameters would actually be the same in-memory instance.  I disabled the cache in the tests which forces disjoint mock objects in the different tests.